### ISSUE TITLE
Splitting out providers from VK core 

### ISSUE THREAD

**Issue by rbitia** [virtual-kubelet/virtual-kubelet#548]
Process *WIP*
1. Figure out if you want to be in the Virtual Kubelet org or move somewhere else. We will create a repo under VK for you. 
2. Move your provider code to the repo under VK
3. Create release lifecycles for your provider
4. Create a release lifecycle for security fixes
5. Maintain your project maintainers in the virtual kubelet community organization and your project. 
6. Figure out a process for triaging issues.

**Comment by sargun** [comment]
I noticed that @cpuguy83 started refactoring the project so outside users could use the CLI more easily, and wrap VK, with their own provider. 

Is the plan to do something else with the current providers?

I started some very rough work on adopting go-plugin (https://github.com/sargun/virtual-kubelet/tree/add-plugin). I imagine that the built-in golang plugin stuff wont work because it does not work on Windows.
